packages/snjs/lib/services/item_manager.ts

```diff
@@ -742,6 +742,23 @@ export class ItemManager extends PureService {
     );
   }
 
+  /**
+   * Returns all parents for a tag
+   * @param tag - The tag for which parents need to be found
+   * @returns Array containing all parent tags
+   */
+  public getParentTags(tag: SNTag): SNTag[] {
```

Comment:
As-is I don't think this function works as expected.

For `foo.bar.tar.zar`, you'd want to return the following parents:

- foo.bar.tar
- foo.bar
- foo

So perhaps you'd return all tags that startWith `foo.`? On the other hand this would return all unrelated children as well, like `foo.bar.tar.sar.zar`. So instead you'd have to just scan the input string itself and construct all possible parent title combinations.

Then you'd name this function something like `getParentChain`, since a tag can only have 1 parent.

Reply:
I've updated the logic, please let me know if it makes sense now, not sure what my brain was thinking the first time around 😆